sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/GatewayConnectionConfig.java

```diff
@@ -119,8 +120,12 @@ public InetSocketAddress getProxy() {
      * @return the {@link GatewayConnectionConfig}.
      */
 
-    public GatewayConnectionConfig setProxy(InetSocketAddress proxy) {
-        this.inetSocketProxyAddress = proxy;
+    public GatewayConnectionConfig setProxy(ProxyOptions proxy) {
+        if (proxy.getType() != ProxyOptions.Type.HTTP) {
```

Comment:
should I throw exception here?

Reply:
Yes, throw `IllegalArgumentException` - with message - only Http Proxy type is supported.

Reply:
Thanks, changed the message to be "Only http proxy type is supported."